src/backend/web/validation.js

```diff
@@ -49,5 +50,27 @@ function validateNewFeed() {
     }
   };
 }
+// Rules for Validation of query params for the /posts route
+const postQueryValidationRules = [
+  query('per_page', 'per_page needs to be empty or a integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+  query('page', 'page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+];
+// create the middle ware for the /posts route with error reporting
+const postQueryValidation = () => {
```

Comment:
Can you change your function name to validatePostsQuery ? I just think it is more action-oriented. Also `Posts` instead of `Post` to reflect the route